build_test/lib/src/utils.dart

```diff
@@ -63,13 +63,58 @@ Future<BuildResult> nextResult(List results) {
   return done.future;
 }
 
+/// Runs build [phases] in a test environment.
+///
+/// The test environment supplies in-memory build [inputs] to the builders under
+/// test. [outputs] may be optionally provided to verify that the builders
+/// produce the expected output.
+///
+/// The keys in [inputs] and [outputs] are paths to file assets and the values
+/// are file contents. The paths must use the following format:
+///
+///     PACKAGE_NAME:PATH_WITHIN_PACKAGE
+///
+/// Where `PACKAGE_NAME` is the name of the package, and `PATH_WITHIN_PACKAGE`
+/// is the path to a file relative to the package. `PATH_WITHIN_PACKAGE` must
+/// include `lib`, `web`, `bin` or `test`. Example: "myapp|lib/utils.dart".
+///
+/// [packageGraph] supplies dependent packages necessary to process [inputs] in
```

Comment:
All this actually does in the in memory asset reader/writer today is define the root package (the one you are allowed to output files in). Anything else is not used.

Ideally it would be able to do more than that though, such as load other packages from disk.